rocketmq/client.py

```diff
@@ -513,3 +513,46 @@ def pull(self, topic, expression='*', max_num=32):
                     pass
                 dll.ReleasePullResult(pull_res)  # NOTE: No need to check ffi return code here
         ffi_check(dll.ReleaseSubscriptionMessageQueue(message_queue))
+
+    def clear_history(self, topic, expression='*'):
```

Comment:
I don't believe `clear_history` is the right API, why not just use `set_message_queue_offset`? You need to manage offset (store it somewhere) by yourself when using `PullConsumer`.

Reply:
say we have this example
`
ready_msgs = circle_consumer.pull(topic)
print(circle_consumer.offset_table)
`
out would be {}


but if we do this
`
ready_msgs = circle_consumer.pull(topic)
for msg in ready_msgs:
        pass
print(circle_consumer.offset_table)
`

Reply:
I will only make a bugfix patch of the _get_mq_key, leave the clean function to be discussed